Gem/Code/Source/Components/NetworkWeaponsComponent.cpp

```diff
@@ -360,10 +363,34 @@ namespace MultiplayerSample
                 const AZ::Vector3& aimAngles = GetNetworkSimplePlayerCameraComponentController()->GetAimAngles();
                 const AZ::Quaternion aimRotation =
                     AZ::Quaternion::CreateRotationZ(aimAngles.GetZ()) * AZ::Quaternion::CreateRotationX(aimAngles.GetX());
-                // TODO: This should probably be a physx raycast out to some maxDistance
                 const AZ::Vector3 fwd = AZ::Vector3::CreateAxisY();
-                const AZ::Vector3 aimTarget = worldTm.GetTranslation() + aimRotation.TransformVector(fwd * 5.0f);
-                AZ::Vector3 aimSource = weaponInput->m_shotStartPosition;
+                AZ::Vector3 baseCameraOffset;
+                AZ::Interface<AZ::IConsole>::Get()->GetCvarValue("cl_cameraOffset", baseCameraOffset);
+                const AZ::Vector3 cameraOffset = aimRotation.TransformVector(baseCameraOffset);
+                
+                // Setup a default aim target
+                AZ::Vector3 aimTarget = worldTm.GetTranslation() + cameraOffset + aimRotation.TransformVector(fwd * sv_WeaponsMaxRange);
+
+                // Cast the ray in the physics system from the center of the camera forward
+                if (auto* sceneInterface = AZ::Interface<AzPhysics::SceneInterface>::Get())
+                {
+                    if (AzPhysics::SceneHandle sceneHandle = sceneInterface->GetSceneHandle(AzPhysics::DefaultPhysicsSceneName);
+                        sceneHandle != AzPhysics::InvalidSceneHandle)
+                    {
+                        AzPhysics::RayCastRequest physicsRayRequest;
+                        physicsRayRequest.m_start = worldTm.GetTranslation() + cameraOffset;
+                        physicsRayRequest.m_direction = aimRotation.TransformVector(fwd);
+                        physicsRayRequest.m_distance = sv_WeaponsMaxRange;
+                        physicsRayRequest.m_queryType = AzPhysics::SceneQuery::QueryType::StaticAndDynamic;
+
+                        AzPhysics::SceneQueryHits result = sceneInterface->QueryScene(sceneHandle, &physicsRayRequest);
+                        if (result)
+                        {
+                            // Set target to first found intersect, if any
```

Comment:
nit: Comment says if any, just want to ensure results if non null will always have one hit. Assume so, but worth double checking.

Reply:
`explicit operator bool() const { return !m_hits.empty(); }`

Yep, it's okay here